File: src/1151-Minimum-Swaps-to-Group-All-1's-Together/1151.go
func minSwaps(data []int) int {
    n := len(data)
    pre := make([]int, n + 1)
    for i := 1; i <= n; i++ {
        pre[i] = pre[i - 1] + data[i - 1]
    }
    
    res, m := pre[n], pre[n]
    for i := m; i < n; i++ {
        res = min(res, m - (pre[i] - pre[i - m]))
    }
    return res
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: src/1151-Minimum-Swaps-to-Group-All-1's-Together/1151.js
var minSwaps = function(data) {
    var n = data.length;
    var pre = Array(n + 1);
    pre.fill(0);
    
    for (var i = 1; i <= n; ++i) {
        pre[i] = pre[i - 1] + data[i - 1];
    }
    var res = pre[n], m = pre[n];
    for (var i = m; i < n; ++i) {
        res = Math.min(res, m - (pre[i] - pre[i - m]));
    }
    return res;
};